pkg/extensions/search/schema.resolvers.go

```diff
@@ -463,6 +463,74 @@ func (r *queryResolver) GlobalSearch(ctx context.Context, query string) (*gql_ge
 	}, nil
 }
 
+// DependencyListForImage is the resolver for the DependencyListForImage field.
+func (r *queryResolver) DerivedImageList(ctx context.Context, image string) ([]*gql_generated.ImageSummary, error) {
+	layoutUtils := common.NewBaseOciLayoutUtils(r.storeController, r.log)
+	imageList := make([]*gql_generated.ImageSummary, 0)
+
+	repoList, err := layoutUtils.GetRepositories()
+	if err != nil {
+		r.log.Error().Err(err).Msg("unable to get repositories list")
+
+		return nil, err
+	}
+
+	if len(repoList) == 0 {
+		r.log.Info().Msg("no repositories found")
+
+		return imageList, nil
+	}
+
+	imageDir, imageTag := common.GetImageDirAndTag(image)
+
+	imageManifest, err := layoutUtils.GetImageManifest(imageDir, imageTag)
+	if err != nil {
+		r.log.Info().Str("image", image).Msg("image not found")
+
+		return imageList, err
+	}
+
+	imageLayers := imageManifest.Layers
+
+	for _, repo := range repoList {
+		repoInfo, err := r.ExpandedRepoInfo(ctx, repo)
+		if err != nil {
+			r.log.Error().Err(err).Msg("unable to get image list")
+
+			return nil, err
+		}
+
+		manifests := repoInfo.Images
+
+		// verify every image
+		for _, manifest := range manifests {
+			layers := manifest.Layers
+
+			sameLayer := 0
+
+			for _, l := range imageLayers {
+				for _, k := range layers {
+					if *k.Digest == string(l.Digest)[7:] {
```

Comment:
Let's use a clean way to do tics check.
layers are of type https://github.com/opencontainers/image-spec/blob/bc9c4bd9b2a4ec6f49f956e193f9c699d854507d/specs-go/v1/descriptor.go#L27
Which is of type Digest, and has various useful methods: https://github.com/opencontainers/go-digest/blob/master/digest.go#L37

Why would we do this comparison relative to the arbitrary characters starting from 7?

Reply:
in order to compare a digest with "sha256:" in front with one which doesn't have it (this is the way it is provided by the function I'm calling to get the digest)